classes/ActionScheduler_Versions.php

```diff
@@ -122,65 +128,24 @@ public static function initialize_latest_version() {
 	 *         'name' => 'Name',
 	 *     ]
 	 *
+	 * @deprecated 3.9.2 Use ActionScheduler_SystemInformation::active_source().
+	 *
 	 * @return array
 	 */
 	public function active_source(): array {
-		$file         = __FILE__;
-		$dir          = __DIR__;
-		$plugins      = get_plugins();
-		$plugin_files = array_keys( $plugins );
-
-		foreach ( $plugin_files as $plugin_file ) {
-			$plugin_path = trailingslashit( WP_PLUGIN_DIR ) . dirname( $plugin_file );
-			$plugin_file = trailingslashit( WP_PLUGIN_DIR ) . $plugin_file;
-
-			if ( 0 !== strpos( dirname( $dir ), $plugin_path ) ) {
-				continue;
-			}
-
-			$plugin_data = get_plugin_data( $plugin_file );
-
-			if ( ! is_array( $plugin_data ) || empty( $plugin_data['Name'] ) ) {
-				continue;
-			}
-
-			return array(
-				'type' => 'plugin',
-				'name' => $plugin_data['Name'],
-			);
-		}
-
-		$themes = (array) search_theme_directories();
-
-		foreach ( $themes as $slug => $data ) {
-			$needle = trailingslashit( $data['theme_root'] ) . $slug . '/';
-
-			if ( 0 !== strpos( $file, $needle ) ) {
-				continue;
-			}
-
-			$theme = wp_get_theme( $slug );
-
-			if ( ! is_object( $theme ) || ! is_a( $theme, \WP_Theme::class ) ) {
-				continue;
-			}
-
-			return array(
-				'type' => 'theme',
-				// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-				'name' => $theme->Name,
-			);
-		}
-
-		return array();
+		_deprecated_function( __METHOD__, '3.9.1', 'ActionScheduler_SystemInformation::active_source()' );
```

Comment:
```suggestion
		_deprecated_function( __METHOD__, '3.9.2', 'ActionScheduler_SystemInformation::active_source()' );
```

Another one that builds on the discussion in [this thread](https://github.com/woocommerce/action-scheduler/pull/1237/files#r1935667047): this version number should be bumped to 3.9.2.